theano/tensor/nnet/abstract_conv2d.py

```diff
@@ -0,0 +1,686 @@
+                             filter_flip = filter_flip)
+    return conv_op(inputs, filters)
+
+
+
+class BaseAbstractConv2d(Op):
+    """Base class for ConvInferace
+
+    FIXME
+    """
+    check_broadcast = False
+    __props__ = ('border_mode', 'subsample')
+
+    def __init__(self,
+                 imshp=None, kshp=None, bsize=None,
+                 border_mode="valid", subsample=(1, 1),
+                 filter_flip = True):
+        if isinstance(border_mode, int):
+            border_mode = (border_mode, border_mode)
+        if isinstance(border_mode, tuple):
+            pad_h, pad_w = map(int, border_mode)
+            border_mode = (pad_h, pad_w)
+        if not ((isinstance(border_mode, tuple) and min(border_mode) >= 0) or
+                border_mode in ('valid', 'full', 'half')):
+            raise ValueError(
+                'invalid border_mode {}, which must be either '
+                '"valid", "full", "half", an integer or a pair of'
+                ' integers'.format(border_mode))
+
+        ### FIXME Check that values are correct
+        self.imshp = imshp
+        self.kshp = kshp
+        self.bsize = bsize
+        self.border_mode = border_mode
+        self.filter_flip = filter_flip
+
+        if len(subsample) != 2:
+            raise ValueError("subsample must have two elements")
+        self.subsample = subsample
+
+    def __str__(self):
+        return '%s{%s, %s}' % (
+            self.__class__.__name__,
+            self.border_mode,
+            str(self.subsample))
+
+    def flops(self, inp, outp):
+        """ Useful with the hack in profilemode to print the MFlops"""
+        # if the output shape is correct, then this gives the correct
+        # flops for any direction, sampling, padding, and border mode
+        inputs, filters = inp
+        outputs, = outp
+        assert inputs[1] == filters[1]
+        # nb mul and add by output pixel
+        flops = filters[2] * filters[3] * 2
+        # nb flops by output image
+        flops *= outputs[2] * outputs[3]
+        # nb patch multiplied
+        flops *= inputs[1] * filters[0] * inputs[0]
+        return flops
+
+
+
+class AbstractConv2d(BaseAbstractConv2d):
+    """
+    FIXME
+    """
+    def __init__(self,
+                 imshp=None,
+                 kshp=None,
+                 bsize=None,
+                 border_mode="valid",
+                 subsample=(1, 1),
+                 filter_flip = True):
+        super(AbstractConv2d, self).__init__(imshp, kshp, bsize,
+                                             border_mode, subsample, filter_flip)
+
+    def make_node(self, img, kern):
+        if img.type.ndim != 4:
+            raise TypeError('img must be 4D tensor')
+        if kern.type.ndim != 4:
+            raise TypeError('kern must be 4D tensor')
+
+        broadcastable=[img.broadcastable[0],
+                       kern.broadcastable[0],
+                       False, False]
```

Comment:
If the output shape can be determined to be 1, broadcastable should be `True` on the corresponding dimension.